ethcore/vm/src/ext.rs

```diff
@@ -63,6 +63,9 @@ pub enum CreateContractAddress {
 
 /// Externalities interface for EVMs
 pub trait Ext {
+	/// Returns the storage value for a given key if reversion happens on the current transaction.
+	fn reverted_storage_at(&self, key: &H256) -> Result<H256>;
```

Comment:
Maybe call it `previous_storage_at` or even `initial_storage_at`? With a comment explaining that it returns the storage value for given key before running current transaction?